.github/workflows/full_test.yml

```diff
@@ -136,136 +160,218 @@ jobs:
+            fast.coverage
+            slow.coverage
+            *.prof.tar.gz
+
+  flaky-test:
+    name: Flaky Tests
+    runs-on: ubuntu-latest
+    env:
+      BUGGER_OFF: "true"
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          repository: ${{ inputs.repo }}
+          ref: ${{ inputs.ref }}
+          token: ${{ secrets.token }}
+          fetch-depth: 0
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+          cache: pip
+          cache-dependency-path: deeplake/requirements/*.txt
+
+      - name: Install requirements
+        run: |
+          pip3 install --upgrade pip --user
+          pip3 install --upgrade setuptools
+          pip3 install -r deeplake/requirements/common.txt
+          pip3 install -r deeplake/requirements/tests.txt
+          pip3 install -r deeplake/requirements/plugins.txt
+          pip3 install -e .[all]
+
+      - name: Run flaky tests
+        id: flaky-tests
+        uses: nick-fields/retry@v2
+        with:
+          timeout_minutes: 20
+          max_attempts: 3
+          shell: bash
+          # Retry seems to only check the last command's exit code, so need to have just one command
+          command: |
+            coverage run --omit="test_*.py" -m pytest --local --hub-cloud --junit-xml=flaky.results.xml --capture=sys -o junit_logging=all -m "flaky" --profile  --pstats-dir=flaky.prof \
+              && mv .coverage flaky.coverage
+#            echo "Exit Code: $?"
+
         env:
-          GITHUB_TOKEN: ${{ secrets.token }}
-          SONAR_TOKEN: ${{ secrets.sonar_token }}
-      
-      - name: Fail if tests are failed
-        if: steps.tests.outcome != 'success'
+          DEEPLAKE_PYTEST_ENABLED: true
+          ENV_HUB_DEV_PASSWORD: ${{ secrets.hub_password }}
+          ACTIVELOOP_HUB_USERNAME: ${{ secrets.hub_username }}
+          ACTIVELOOP_HUB_PASSWORD: ${{ secrets.hub_password }}
+
+      - name: Process Results
+        if: always()
         run: |
-          exit 1
+          tar -czf flaky.prof.tar.gz flaky.prof
 
-      - name: Run backwards compatibility tests
+      - name: Save Test Results
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: test-results-flaky_${{ github.run_number }}-${{ github.run_attempt }}
+          path: |
+            flaky.results.xml
+            flaky.coverage
+            flaky.prof.tar.gz
+
+  buh-test:
+    name: Backwards Compatibility Test
+    runs-on: ubuntu-latest
+    timeout-minutes: 120
+    env:
+      BUGGER_OFF: "true"
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          repository: ${{ inputs.repo }}
+          ref: ${{ inputs.ref }}
+          token: ${{ secrets.token }}
+          fetch-depth: 0
+
+      - name: Checkout the buH source code
+        uses: actions/checkout@v3
+        with:
+          path: buH
+          ref: cache_datasets # Remove after merged
```

Comment:
This should be fixed before merging to `main`, but needs [activeloopai/buH#7] to be merged first

Reply:
removed